aptos-move/aptos-aggregator/src/delta_change_set.rs

```diff
@@ -233,6 +235,34 @@ impl DeltaChangeSet {
         // All deltas are applied successfully.
         Ok(WriteSetMut::new(materialized_write_set))
     }
+
+    pub fn merge_with(self, other: Self) -> PartialVMResult<Self> {
+        let mut buffer = vec![];
+        let mut index_map = BTreeMap::new();
+
+        for (key, op) in self
+            .delta_change_set
+            .into_iter()
+            .chain(other.delta_change_set)
+        {
+            match index_map.entry(key) {
+                btree_map::Entry::Occupied(entry) => {
+                    let idx = *entry.get();
+                    let elem: &mut (StateKey, DeltaOp) = &mut buffer[idx];
+                    let op_mut: &mut DeltaOp = &mut elem.1;
+                    op_mut.merge_with(op)?;
+                }
+                btree_map::Entry::Vacant(entry) => {
+                    buffer.push((entry.key().clone(), op));
+                    entry.insert(buffer.len());
+                }
+            }
+        }
+
+        Ok(Self {
+            delta_change_set: buffer,
+        })
+    }
```

Comment:
@gelash @georgemitenkov Please let me know if this looks good to you. Right now I'm preserving the ordering of the `DeltaOp`s, but I'm not really sure if this is necessary.

Also thoughts on changing the representation of `DeltaChangeSet` from `Vec` to `BTreeMap`?

Reply:
Order surely doesn't matter, and HashMap is fine by me, we iterate over it in parallel execution, but shouldn't become much less efficient?

@georgemitenkov, thoughts?

Reply:
I don't think ordering matters, HashMap with StateKeys makes more sense I guess

Reply:
As long as CryptoHash::hash() yields deterministally for a HashMap (which I believe so). But I'm glad to see we are using BTreeMaps.